File: app/lib/network/handlers/order_client_handler.dart
import 'dart:convert';
import '/models/dto/order_dto.dart';
import '/models/dto/simple_response.dart';
import '/network/network_constants.dart';
import '/providers/kitchen_provider.dart';
import 'base_handler.dart';

class OrderClientHandler extends BaseHandler {
  final KitchenProvider kitchenProvider;

  OrderClientHandler(this.kitchenProvider);

  @override
  String get name => NetworkConstants.orderService;

  @override
  Future<dynamic> process(String action, String payload) async {
    switch(action) {
      case NetworkConstants.orderCompletedAction:
        final order = OrderDTO.fromJson(jsonDecode(payload));
        if (order.id != null) {
          kitchenProvider.removeOrder(order.id!);
        }
        break;
      case NetworkConstants.orderCreatedAction:
        final order = OrderDTO.fromJson(jsonDecode(payload));
        kitchenProvider.addOrder(order);
        break;
      default:
        throw Exception("Action $action not implemented");
    }
    return SimpleResponseDTO(message: "OK");
  }
}

File: local/src/main/java/com/concessions/local/kitchen/network/OrderClientHandler.java
package com.concessions.local.kitchen.network;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Component;

import com.concessions.common.network.AbstractHandler;
import com.concessions.common.network.NetworkException;
import com.concessions.dto.OrderDTO;
import com.concessions.local.kitchen.model.OrderDisplayModel;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

@Component
@ConditionalOnProperty(
	    value = "local.network.client", 
	    havingValue = "true", 
	    matchIfMissing = false
	)
public class OrderClientHandler extends AbstractHandler {

	protected OrderDisplayModel orderDisplayModel;
	
	public OrderClientHandler (@Autowired ObjectMapper mapper, @Autowired OrderDisplayModel orderDisplayModel) {
		super(mapper);
		this.orderDisplayModel = orderDisplayModel;
	}

	@Override
	public String getName() {
		return ORDER_SERVICE;
	}

	@Override
	public Object process(String action, String payload) throws NetworkException {
		switch (action) {
		case ORDER_COMPLETED_ACTION:
			processOrderCompleted(payload);
			break;
		case ORDER_CREATED_ACTION:
			processOrderCreated(payload);
			break;
		}
		return success;
	}

	protected void processOrderCompleted (String payload) {
		try {
			OrderDTO order = mapper.readValue(payload, OrderDTO.class);
			orderDisplayModel.removeOrder(order);
		} catch (JsonProcessingException ex) {
			ex.printStackTrace();
		}
	}
	
	protected void processOrderCreated (String payload) {
		try {
			OrderDTO order = mapper.readValue(payload, OrderDTO.class);
			orderDisplayModel.addOrder(order);
		} catch (JsonProcessingException ex) {
			ex.printStackTrace();
		}
	}
}
